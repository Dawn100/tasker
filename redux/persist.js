import { AsyncStorage } from "react-native";

export const loadState = async () => {
  try {
    const taskerStore = await AsyncStorage.getItem("TASKERTODOS");
    if (taskerStore !== null) {
      return JSON.parse(taskerStore);
    }

    console.log("load state returned errr-> " + error);
    return undefined;
  } catch (error) {
    console.log("load state error " + error);
    return undefined;
  }
};

export const saveStore = async store => {
  try {
    let storingValue = JSON.stringify({
      todoStore: store,
      time: new Date()
    });
    await AsyncStorage.setItem("TASKERTODOS", storingValue);
  } catch (err) {
    console.log("Save store error" + err);
  }
};
