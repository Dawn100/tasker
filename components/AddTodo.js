import React, { Component } from "react";
import { View, TextInput, Button } from "react-native";
import { connect } from "react-redux";
import { addTodo } from "../redux/actions";

class AddTodo extends Component {
  constructor(props) {
    super(props);
    this.state = {
      todo: ""
    };
  }
  render() {
    return (
      <View testID={"form"}>
        <TextInput
          testID={"textInput"}
          placeholder="Add a Task..."
          value={this.state.todo}
          onChangeText={todo => {
            this.setState({ todo });
          }}
        />
        <Button
          testID="addButton"
          onPress={() => this.props.onAdd(this.state.todo)}
          title="Add"
        />
      </View>
    );
  }
}

const mapDispatchToProps = dispatch => {
  return {
    onAdd: todo => {
      dispatch(addTodo(todo));
    }
  };
};

export default connect(
  null,
  mapDispatchToProps
)(AddTodo);
