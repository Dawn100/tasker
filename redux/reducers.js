import { combineReducers } from "redux";
import { ADD_TODO, DELETE_TODO } from "./actions";

const todosReducer = (state = [], action) => {
  switch (action.type) {
    case ADD_TODO:
      return [
        {
          todo: action.payload.todo,
          completed: false,
          id: state.reduce((maxId, todo) => Math.max(todo.id, maxId), -1) + 1
        },
        ...state
      ];
    case DELETE_TODO:
      state.map((todo, index) => {
        if (todo.id === action.payload.id) {
          state.splice(index, 1);
        }
      });
      return state;

    default:
      return state;
  }
};

export const reducer = combineReducers({
  todos: todosReducer
});
