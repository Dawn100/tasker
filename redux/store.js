import { createStore, applyMiddleware } from "redux";
import { reducer } from "./reducers";
import { loadState,saveStore } from "./persist";
import { createLogger } from "redux-logger";

const persistedState=loadState()
console.log(persistedState)

const store = createStore(reducer,persistedState.todoStore);

store.subscribe(()=>{
	saveStore(store.getState())
})


export default store