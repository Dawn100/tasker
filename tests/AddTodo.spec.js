import React from "react";
import { Provider } from "react-redux";

import AddTodo from "../components/AddTodo";
import { create } from "react-test-renderer";
import { store } from "../redux/store";

describe("AddTodo Component", function() {
  it("Renders self and children", () => {
    const add = create(
      <Provider store={store}>
        <AddTodo onAdd={() => {}} />
      </Provider>
    );
    expect(add).toMatchSnapshot();
    add.root.findByProps({ testID: "addButton" }).props.onPress();
  });
});
