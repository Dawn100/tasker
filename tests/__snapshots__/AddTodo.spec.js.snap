// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo Component Renders self and children 1`] = `
<View
  testID="form"
>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Add a Task..."
    rejectResponderTermination={true}
    testID="textInput"
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID="addButton"
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Add
      </Text>
    </View>
  </View>
</View>
`;
